File: YTMSiparis/PremiumUyg/pyhton/Siparis.py
class Siparis:
    @staticmethod
    def odeme_yap(kullanici):
        toplam_tutar = 0

        # Python'da döngü
        for urun in kullanici.urunler:
            toplam_tutar += urun.fiyat  # fiyat özelliği kullanılıyor

        toplam_tutar = kullanici.get_toplam_tutar(toplam_tutar)
        print(f"{type(kullanici).__name__} Toplam Tutar = {toplam_tutar}")

File: YTMSiparis/PremiumUyg/c#/OCP/OCP/Siparis.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OCP
{
    public class Siparis
    {
        public static void OdemeYap(Kullanici kullanici)
        {
            double toplamTutar = 0;

            // C#'ta foreach döngüsü
            foreach (Urun urun in kullanici.Urunler)
            {
                toplamTutar += urun.Fiyat; // Fiyat özelliği kullanılıyor
            }

            toplamTutar = kullanici.GetToplamTutar(toplamTutar);
            Console.WriteLine($"{kullanici.GetType().Name} Toplam Tutar = {toplamTutar}");
        }
    }
}
